ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadRequestQueue.java

```diff
@@ -103,9 +101,20 @@ public DownloadRequestQueue() {
 	 * Deprecated:
 	 */
 	public DownloadRequestQueue(int threadPoolSize) {
-		mDelivery = new CallBackDelivery(new Handler(Looper.getMainLooper()));
-		int processors = Runtime.getRuntime().availableProcessors();
-		mDownloadDispatchers = new DownloadDispatcher[processors];
+		doConstruct(new Handler(Looper.getMainLooper()));
+	}
+
+	/**
+	 * Construct with provided callback handler.
+	 *
+	 * @param callbackHandler
+	 */
+	public DownloadRequestQueue(Handler callbackHandler) {
+		if(callbackHandler == null) {
+			callbackHandler = new Handler(Looper.getMainLooper());
```

Comment:
This CSTR should thrown an exception if null is passed in. Could you tell the method to throw "Invalid exception" and throw a new exception with meaning ful message.